src/lavinmq/mqtt/session.cr

```diff
@@ -131,6 +133,20 @@ module LavinMQ
         raise ClosedError.new(cause: ex)
       end
 
+      def build_packet(env, packet_id) : MQTT::Publish
+        msg = env.message
+        retained = msg.properties.try &.headers.try &.["x-mqtt-retain"]? == true
+        qos = msg.properties.delivery_mode || 0u8
+        MQTT::Publish.new(
+          packet_id: packet_id,
+          payload: msg.body,
+          dup: env.redelivered,
+          qos: qos,
+          retain: retained,
+          topic: msg.routing_key.tr(".", "/")
```

Comment:
@spuun is this `.tr`  really correct?

Reply:
I don't think you need to between topic and routing key here. I think we're storing it as topic. 
But if it's a routing key you want to convert `.` to `/`.

Reply:
yep thats a rest that i forgot to remove

Reply:
removing the `tr` then. 

WDYT of the change otherwise?

Reply:
looks good :)